cabal-install/Distribution/Client/SetupWrapper.hs

```diff
@@ -334,7 +335,8 @@ selfExecSetupMethod verbosity options _pkg bt mkargs = do
 
   searchpath <- programSearchPathAsPATHVar
                 (getProgramSearchPath (useProgramConfig options))
-  env        <- getEffectiveEnvironment [("PATH", Just searchpath)]
```

Comment:
For this to work properly with the self-exec setup method we probably also want `HASKELL_DIST_DIR=/foo/bar cabal $CMD` to behave identically to `cabal $CMD --builddir=/foo/bar`.